front/src/modules/rollingStock/components/RollingStockEditor/RollingStockEditorFormHelpers.tsx

```diff
@@ -372,3 +374,87 @@ export const RollingStockEditorOnboardSystemEquipmentForm = ({
     </div>
   );
 };
+
+export const RollingStockEditorCategoryForm = ({
+  rollingStockValues,
+  setRollingStockValues,
+}: RollingStockEditorParameterFormProps) => {
+  const { t } = useTranslation(['rollingstock']);
+
+  type CategoryOption = { id: RollingStockCategory | ''; label: string };
+  const categoryOptions: CategoryOption[] = [
+    ...Object.values(RollingStockCategoryDict).map((category) => ({
+      id: category,
+      label: t(`categoriesOptions.${category}`),
+    })),
+    { id: '', label: '' },
+  ];
+
+  const handlePrimaryCategoryChange = (selectedCategory: CategoryOption | undefined) => {
+    if (!selectedCategory) return;
+    setRollingStockValues((prevValues) => ({
+      ...prevValues,
+      primaryCategory: selectedCategory.id || undefined,
+      categories: selectedCategory.id
+        ? [...rollingStockValues.categories, selectedCategory.id]
+        : rollingStockValues.categories,
+    }));
+  };
```

Comment:
When updating a primary category, I don't think we can select an empty one
```suggestion
  const handlePrimaryCategoryChange = (selectedCategory: CategoryOption) => {
    setRollingStockValues((prevValues) => ({
      ...prevValues,
      primaryCategory: selectedCategory.id,
      categories: [...rollingStockValues.categories, selectedCategory.id]
    }));
  };
```

Reply:
Sadly the parameter onChange of the select component has the following signature : onChange: (e?: T | undefined) => void, where T is the type of the provided options. So even if if it should not happen (though I did not check the implementation of select to know if perhaps it happens when handling some kind of events), we have to deal with the possibility for typescript to be happy.
 
 There is another check here though for the truth value of selectedCategory.id (in primaryCategory: selectedCategory.id || undefined), which happens when selecting the blank value back for the primary category. We could absolutely disable this by removing the blank string from the options as long as a value was selected for primary category. Would you like me to make this change? I'll ask Thibaut what he thinks

Reply:
So actually select has a weird behavior, where it sends undefined instead of the selected option if the selected option.id is falsy... Well,at least it should serves to simplify the code a bit. 

I also discussed with Thibaut and now we have a Choose label.

Reply:
Looks great !